resource-sharing-partners/src/test/java/no/sikt/rsp/ResourceSharingPartnerTest.java

```diff
@@ -264,6 +278,127 @@ public void shouldExtractCertainDataIfAlmaOrBibsysLibrary(String katsys) throws
         //TODO:LocateProfile in partnerDetails should be set when isAlmaOrBibsys, otherwise null. SMILE-1573
     }
 
+    @ParameterizedTest(name = "Should handle katsys codes differently")
+    @ValueSource(strings = {ALMA, BIBSYS, TIDEMANN})
+    public void shouldExtractPartnerDetailsProfileDataCorrectly(final String katsys) throws IOException {
+        final Record firstRecord = new RecordBuilder(BigInteger.ONE, LocalDate.now(), katsys)
+                                       .withBibnr("bibnr1")
+                                       .withLandkode("landkode1")
+                                       .withEpostAdr("adr@example.com")
+                                       .withEressurser(new EressurserBuilder().build())
+                                       .build();
+
+        final Record secondRecord = new RecordBuilder(BigInteger.valueOf(2), LocalDate.now(), katsys)
+                                        .withBibnr("bibnr2")
+                                        .withLandkode("landkode2")
+                                        .withEpostBest("best@example.com")
+                                        .withEressurser(
+                                            new EressurserBuilder().withNncipUri(NNCIP_SERVER).build())
+                                        .build();
+
+        final Record thirdRecord = new RecordBuilder(BigInteger.valueOf(3), LocalDate.now(), katsys)
+                                        .withBibnr("bibnr3")
+                                        .withLandkode("landkode3")
+                                        .build();
+
+        final Record fourthRecord = new RecordBuilder(BigInteger.valueOf(4), LocalDate.now(), katsys)
+                                       .withBibnr("bibnr4")
+                                       .withLandkode("landkode4")
+                                       .withEpostBest("invalid")
+                                       .withEpostAdr("invalid")
+                                       .build();
+
+        var basebibliotekGenerator = new BasebibliotekGenerator(firstRecord, secondRecord, thirdRecord, fourthRecord);
+        var basebibliotek = basebibliotekGenerator.generateBaseBibliotek();
+        var basebibliotekXml = BasebibliotekGenerator.toXml(basebibliotek);
+        var uri = s3Driver.insertFile(randomS3Path(), basebibliotekXml);
+        var s3Event = createS3Event(uri);
+
+        when(mockedEnvironment.readEnv(ILL_SERVER_ENVIRONMENT_NAME)).thenReturn(ILL_SERVER_ENVIRONMENT_VALUE);
+
+        resourceSharingPartnerHandler.handleRequest(s3Event, CONTEXT);
+
+        var partners = resourceSharingPartnerHandler.getPartners();
+
+        int partnerIdx = 0;
+        for (Partner partner : partners) {
```

Comment:
Kan dette bli konvertert til noen private metoder?

Reply:
Jeg refaktorer testen slik at vi får flere enklere tester for lesbarhetens skyld.

Reply:
Refaktorerte testene